### DISCUSSION TITLE
How to avoid push a specific universal link with expo-router?

### DISCUSSION THREAD

**Discussion by jaybowang** [expo/expo#25153] [Archived: Questions]
I use `native-wechat` for login with WeChat app.
It navigates from WeChat app back to my app with a universal link like `https://app.domain.com/app/wechat-appId/oauth?code=codeFromWeChatPlatform&state=`.
The problem is that expo-router automatically pushes to `/app/wechat-appId/oauth` which is a undefined route.
What I really need is do not push to any routes.

I checked the docs and haven't got any idea. Any comments is highly appreciated.

**Comment by newerton** [comment]
I have the same problem in a personal project.

The user is on the registration screen /app/(register)/email, he will receive an email with the link https://domain.com/validation/email, in the project there is no such route.

**Comment by nicolopadovan** [comment]
I have the exact same problem

**Comment by jaybowang** [comment]
The solution is simple. Just create those missing folders and a _layout file.

For example https://domain.com/validation/email
```javascript
// step one: create a folder named validation, create a subfolder named email.

// step two: create a _layout.tsx file in the email folder
// ./app/validation/email/_layout.tsx

import { Stack } from 'expo-router/stack';

export default function MyLayout() {
  return (
    <Stack>
      <Stack.Screen
        name='my-target-route' 
        // or href='..'
        options={{ animation: 'none' }}
      />
    </Stack>
  );
}

```

**Reply by newerton** [reply]
@jaybowang This is already in the documentation.

It becomes impossible to create a file for each Universal Link URL.

In React Navigation, you have the NavigationContainer, which allows you to make this configuration.

```ts
<NavigationContainer
  linking={{
    prefixes: ['https://mychat.com', 'mychat://'],
    config: {
      screens: {
        Chat: 'feed/:sort',
      },
    },
  }}
>
  {/* content */}
</NavigationContainer>
```

https://reactnavigation.org/docs/navigation-container/#linking

**Reply by davidoluseun** [reply]
@newerton could you provide a snippet to achieve this in Expo Router?

**Reply by newerton** [reply]
@davidoluseun You can't run it by configuring a file, you'll need to create the Universal Link routes, and use Redirect.

**Reply by davidoluseun** [reply]
Oh, thank you for your response! I was hoping to configure this  in Expo Router without needing to use Redirect

**Comment by dyarfaradj** [comment]
Any update on this? I'm using react-native-track-player and when clicking on the media in controlcenter I get this event: 
{"url": "trackplayer://notification.click"}

I'd like to prevent expo from routing to anything

**Comment by pladaria** [comment]
https://docs.expo.dev/router/advanced/native-intent/

To disable automatic deep link handling, create a `+native-intent.tsx` file with this content:

```tsx
export const redirectSystemPath = () => {
  return false;
};
```

You could implement your deepLink logic inside that function or create a custom Hook somewhere else that handles `Linking.addEventListener("url")`  events and/or `Linking.getInitialURL`

**Reply by jensneuber** [reply]
@pladaria Thanks for the hint!

I had to use 
```typescript
export const redirectSystemPath = () => {
  return '/';
};
```
though, or my app would crash. Maybe this helps someone
